e2e_tests/tests/cluster/test_logging.py

```diff
@@ -105,13 +105,36 @@ def task_log_fields(follow: Optional[bool] = None) -> Iterable[LogFields]:
         return bindings.get_TaskLogsFields(session, taskId=task_id, follow=follow)
 
     try:
-        check_logs(
-            log_regex,
-            functools.partial(api.task_logs, session, task_id),
-            functools.partial(bindings.get_TaskLogsFields, session, taskId=task_id),
-        )
-    except socket.timeout:
-        raise TimeoutError(f"timed out waiting for {task_type} with id {task_id}")
+        result: Any = None
+
+        def do_check_logs() -> None:
+            nonlocal result
+            try:
+                check_logs(
+                    log_regex,
+                    functools.partial(api.task_logs, session, task_id),
+                    functools.partial(bindings.get_TaskLogsFields, session, taskId=task_id),
+                )
+                # Success!
+                result = True
+            except Exception as e:
+                result = e
+
+        thread = threading.Thread(target=do_check_logs, daemon=True)
+        thread.start()
+        thread.join(timeout=5 * 60)
+        if thread.is_alive():
+            # The thread did not exit
+            raise ValueError("do_check_logs thread did not exit")
```

Comment:
If it didn't exit, doesn't that mean that we timed out? Shouldn't we say that instead if that's the case?

Reply:
There might be value to printing stdout in this case.

Reply:
@MikhailKardash what do you mean?  Isn't that what the `except Exception` clause does, is print the stdout/stderr from the task?